selfdrive/car/tesla/carcontroller.py

```diff
@@ -785,3 +646,165 @@ def update(self, enabled, CS, frame, actuators, \
+      self.lead2Dy = 0.
+      self.lead2Class = 0
+      self.lead2Id = 0
+      self.lead2Vx = 0xF
+    messages.append(teslacan.create_DAS_LR_object_msg(0,self.leadClass, self.leadId,
+          self.leadDx,self.leadDy,self.leadVx,self.lead2Class,
+          self.lead2Id,self.lead2Dx,self.lead2Dy,self.lead2Vx))
+    return messages
+
+  def handlePathPlanSocketForCurvatureOnIC(self, pathPlanSocket, CS):
+    pp = messaging.recv_one(pathPlanSocket).pathPlan
+    if pp.paramsValid:
+      if pp.lProb > 0.75:
+        self.lLine = 3
+      elif pp.lProb > 0.5:
+        self.lLine = 2
+      elif pp.lProb > 0.25:
+        self.lLine = 1
+      else:
+        self.lLine = 0
+      if pp.rProb > 0.75:
+        self.rLine = 3
+      elif pp.rProb > 0.5:
+        self.rLine = 2
+      elif pp.rProb > 0.25:
+        self.rLine = 1
+      else:
+        self.rLine = 0
+      #first we clip to the AP limits of the coefficients
+      self.curv0 = -clip(pp.cPoly[3],-3.5,3.5) #self.curv0Matrix.add(-clip(pp.cPoly[3],-3.5,3.5))
+      self.curv1 = -clip(pp.cPoly[2],-0.2,0.2) #self.curv1Matrix.add(-clip(pp.cPoly[2],-0.2,0.2))
+      self.curv2 = -clip(pp.cPoly[1],-0.0025,0.0025) #self.curv2Matrix.add(-clip(pp.cPoly[1],-0.0025,0.0025))
+      self.curv3 = -clip(pp.cPoly[0],-0.00003,0.00003) #self.curv3Matrix.add(-clip(pp.cPoly[0],-0.00003,0.00003))
+      self.laneWidth = pp.laneWidth
+      self.laneRange = 50 # it is fixed in OP at 50m pp.viewRange
+      self.visionCurvC0 = self.curv0
+      self.prev_ldwStatus = self.ldwStatus
+      self.ldwStatus = 0
+      if (self.ALCA.laneChange_direction != 0) and pp.alcaError:
+        self.ALCA.stop_ALCA()
+      if self.alca_enabled:
+        #exagerate position a little during ALCA to make lane change look smoother on IC
+        if self.ALCA.laneChange_over_the_line:
+          self.curv0 = self.ALCA.laneChange_direction * self.laneWidth - self.curv0
+        self.curv0 = clip(self.curv0, -3.5, 3.5)
+      else:
+        if CS.enableLdw and (not CS.blinker_on) and (CS.v_ego > 15.6) and (not CS.steer_override):
+          if pp.lProb > LDW_LANE_PROBAB:
+            lLaneC0 = -pp.lPoly[3]
+            if abs(lLaneC0) < LDW_WARNING_2:
+              self.ldwStatus = 3
+            elif  abs(lLaneC0) < LDW_WARNING_1:
+              self.ldwStatus = 1
+          if pp.rProb > LDW_LANE_PROBAB:
+            rLaneC0 = -pp.rPoly[3]
+            if abs(rLaneC0) < LDW_WARNING_2:
+              self.ldwStatus = 3
+            elif  abs(rLaneC0) < LDW_WARNING_1:
+              self.ldwStatus = 1
+      if not(self.prev_ldwStatus == self.ldwStatus):
+        self.warningNeeded = 1
+        if self.ldwStatus > 0:
+          self.warningCounter = 50
+    else:
+      self.lLine = 0
+      self.rLine = 0
+      self.curv0 = self.curv0Matrix.add(0.)
+      self.curv1 = self.curv1Matrix.add(0.)
+      self.curv2 = self.curv2Matrix.add(0.)
+      self.curv3 = self.curv3Matrix.add(0.)
+
+  # Bog: What would a good name for this be?
```

Comment:
Raf, this function send the Left and Right radar identified cars from the radard to the IC

Reply:
Cool, I'll rename to:
```python
def showLeftAndRightCarsOnICCanMessages(self, icCarLRSocket):
```